src/main/java/com/zenith/network/client/handler/incoming/SetTimeHandler.java

```diff
@@ -1,15 +1,18 @@
 package com.zenith.network.client.handler.incoming;
 
 import com.github.steveice10.mc.protocol.packet.ingame.clientbound.level.ClientboundSetTimePacket;
+import com.zenith.cache.data.chunk.WorldTimeData;
 import com.zenith.network.client.ClientSession;
 import com.zenith.network.registry.AsyncPacketHandler;
 
+import static com.zenith.Shared.CACHE;
 import static com.zenith.Shared.TPS_CALCULATOR;
 
 public class SetTimeHandler implements AsyncPacketHandler<ClientboundSetTimePacket, ClientSession> {
 
     @Override
     public boolean applyAsync(ClientboundSetTimePacket packet, ClientSession session) {
+        CACHE.getChunkCache().setWorldTimeData(new WorldTimeData(packet));
```

Comment:
This packet is received like every second by default, I'd prefer to avoid the GC overhead of constructing more objects this frequently.

I think we can just maintain a state that's updated in-place through setters.